src/modules/distributions.ts

```diff
@@ -103,6 +119,71 @@ export function getMinMaxFromBucket(bucketLabel: string): number[] {
   return [Number(min.trim()), Number(max.trim())];
 }
 
+/**
+ * Calculates the distribution of an array of objects grouped
+ *
+ * @export
+ * @param {IBucket[]} buckets
+ * @param {Record<string, unknown[]>} dataGrouped
+ * @return {ISerieDistribution} The distribution with labels and data
+ */
+export function calcDistributionWithSeries(
+  buckets: IBucket[],
+  dataGrouped: Record<string, unknown[]>,
+): ISerieDistribution {
+  let eventsTotal = 0;
+
+  const data = Object.entries(dataGrouped).map(([key, value]) => {
+    let eventsSerie = 0;
+
+    const serieName = key;
+    const dataVal: Record<string, unknown>[] = [];
+
+    value.forEach((v) => {
+      buckets.forEach((d) => {
+        const valObj = (v as Record<string, number | string>).data as number;
+
+        if (d.inside(valObj)) {
+          dataVal.push({
+            data: valObj,
+            interval: d.label,
+          });
+        }
+      });
+
+      eventsTotal++;
+    });
+
+    const valuesGrouped = countBy(dataVal, 'interval');
+
+    Object.values(valuesGrouped).forEach((v) => {
+      eventsSerie += v;
+    });
+
+    const dataArr = new Array(buckets.length).fill(0);
+
+    Object.entries(valuesGrouped).forEach(([keyV, valueV]) => {
+      const idx = buckets.findIndex((l) => l.label === keyV);
+
+      dataArr[idx] = valueV;
+    });
+
+    return {
+      name: serieName,
+      count: dataArr,
+      percentage_serie: dataArr.map((d) => (d * 100) / eventsSerie, 2),
```

Comment:
You can use the https://github.com/alrico88/math-helper-functions#calcpercent from this library